espflash/src/error.rs

```diff
@@ -469,19 +469,19 @@ impl InvalidSubTypeError {
 }
 
 #[derive(Debug, Error, Diagnostic)]
-#[error("No factory app partition was found")]
+#[error("No app partition was found")]
 #[diagnostic(
-    code(espflash::partition_table::no_factory_app),
-    help("Partition table must contain a factory app partition")
+    code(espflash::partition_table::no_app),
```

Comment:
One question: I'm not familiar with the `diagnostic` macro, is this change valid?